src/Salt/SiteBundle/Controller/CommentsController.php

```diff
@@ -0,0 +1,215 @@
+        $comment = new Comment();
+        $em = $this->getDoctrine()->getManager();
+
+        $itemId = $request->request->get('itemId');
+        $itemType = $request->request->get('itemType');
+        $parentId = $request->request->get('parent');
+
+        if ($this->existItem($itemId, $itemType)) {
+            $comment->setContent(trim($request->request->get('content')));
+            $comment->setUser($user);
+            $comment->setFullname($user->getUsername().' - '.$user->getOrg()->getName());
+            $comment->setItem($itemType.':'.$itemId);
+
+            if (!empty($parentId) && filter_var($parentId, FILTER_VALIDATE_INT)) {
+                $parent = $em->getRepository('SaltSiteBundle:Comment')->findById($parentId);
+                $comment->setParent(($parent)?$parentId:null);
+            } else {
+                $comment->setParent(null);
+            }
+
+            $em->persist($comment);
+            $em->flush();
+
+            $response = $this->apiResponse($comment);
+            return $response;
+        }
+
+        return new Response('Item not found', 404);
+    }
+
+    /**
+     * @Route("/comments/{itemId}/{itemType}", name="get_comments")
+     *
+     * @Method("GET")
+     *
+     * @Security("is_granted('view_comment')")
+     */
+    public function listAction($itemId, $itemType, UserInterface $user = null)
+    {
+        $em = $this->getDoctrine()->getManager();
+        $comments = $em->getRepository('SaltSiteBundle:Comment')->findByItem($itemType.':'.$itemId);
+
+        if ($user) {
+            foreach ($comments as $comment){
+                if ($comment->getUser()->getId() == $user->getId()){
+                    $comment->setCreatedByCurrentUser(true);
+                }
+
+                $upvotes = $comment->getUpvotes();
+
+                foreach ($upvotes as $upvote) {
+                    if ($upvote->getUser()->getId() == $user->getId()) {
+                        $comment->setUserHasUpvoted(true);
+                        break;
+                    }
+                }
+            }
+        }
+
+        $response = $this->apiResponse($comments);
+        return $response;
+    }
+
+    /**
+     * @Route("/comments/{id}")
+     *
+     * @Method("PUT")
+     *
+     * @Security("is_granted('comment')")
+     */
+    public function updateAction(Comment $comment, Request $request, UserInterface $user)
+    {
+        $em = $this->getDoctrine()->getManager();
+
+        $comment->setContent($request->request->get('content'));
+        $em->persist($comment);
+        $em->flush($comment);
+
+        $response = $this->apiResponse($comment);
+        return $response;
+    }
+
+    /**
+     * @Route("/comments/delete/{id}")
+     *
+     * @Method("DELETE")
+     *
+     * @Security("is_granted('comment')")
+     */
+    public function deleteAction(Comment $comment, UserInterface $user)
+    {
+        $em = $this->getDoctrine()->getManager();
+
+        $em->remove($comment);
```

Comment:
Can any user remove the comment, or does the `$comment->getUser()` need to match `$user`?